packages/dev/loaders/src/glTF/glTFFileLoader.ts

```diff
@@ -48,11 +48,11 @@ function readAsync(arrayBuffer: ArrayBuffer, byteOffset: number, byteLength: num
 
 function readViewAsync(arrayBufferView: ArrayBufferView, byteOffset: number, byteLength: number): Promise<Uint8Array> {
     try {
-        if ((arrayBufferView as Uint8Array).byteOffset + byteLength > arrayBufferView.byteLength) {
+        if (arrayBufferView.byteOffset + byteOffset + byteLength > arrayBufferView.buffer.byteLength) {

```

Comment:
I think this check isn't correct anymore. `arrayBufferView.byteLength` is a smaller number than `arrayBufferView.buffer.byteLength`?

Reply:
Yes, `arrayBufferView.byteLength` is smaller than `arrayBufferView.buffer.byteLength`, as `arrayBufferView.buffer` is the underlying buffer the view is built upon, and `arrayBufferView.buffer.byteLength` is the total size of this buffer. At most, `arrayBufferView.byteLength == arrayBufferView.buffer.byteLength` (possible only if `arrayBufferView.byteOffset == 0`).

`arrayBufferView.byteOffset` is the offset inside `arrayBufferView.buffer`, so:

`arrayBufferView.byteOffset + byteOffset + byteLength > arrayBufferView.buffer.byteLength`

checks that we won't go past the end of the underlying buffer when we create the new view.

But that may not be the right thing to do, as you explained me in our chat.